sdk/program/src/lib.rs

```diff
@@ -1,3 +1,650 @@
+//!   - Instruction: [`solana_program::loader_instruction`]
+//!
+//! # Sysvars
+//!
+//! Sysvars are special accounts that contain dynamically updating data about
+//! the network cluster, the blockchain history, and the executing transaction.
+//!
+//! The program IDs and deserializers for sysvars are defined in the [`sysvar`]
+//! module, and simple sysvars implement the [`Sysvar`] trait. Since to a Solana
+//! program sysvars are just accounts, if the `AccountInfo` is provided to the
+//! program, then the program can deserialize the sysvar with
+//! [`Sysvar::from_account_info`] to access its data, as in this example that
+//! logs the [`clock`][clk] sysvar.
+//!
+//! [`Sysvar`]: sysvar::Sysvar
+//! [`Sysvar::from_account_info`]: sysvar::Sysvar::from_account_info
+//! [clk]: sysvar::clock
+//!
+//! ```
+//! use solana_program::{
+//!     account_info::{next_account_info, AccountInfo},
+//!     clock,
+//!     entrypoint::ProgramResult,
+//!     msg,
+//!     pubkey::Pubkey,
+//!     sysvar::Sysvar,
+//! };
+//!
+//! fn process_instruction(
+//!     program_id: &Pubkey,
+//!     accounts: &[AccountInfo],
+//!     instruction_data: &[u8],
+//! ) -> ProgramResult {
+//!     let account_info_iter = &mut accounts.iter();
+//!     let clock_account = next_account_info(account_info_iter)?;
+//!     let clock = clock::Clock::from_account_info(&clock_account)?;
+//!     msg!("clock: {:#?}", clock);
+//!     Ok(())
+//! }
+//! ```
+//!
+//! As an optimization, to avoid including sysvar accounts in instructions and deserializing them
+//! in the program, some sysvars implement [`Sysvar::get`], which loads
+//! a deserialized sysvar directly from the runtime, as in this
+//! example, again using the `clock` sysvar:
+//!
+//! [`Sysvar::get`]: sysvar::Sysvar::get
+//!
+//! ```
+//! use solana_program::{
+//!     account_info::AccountInfo,
+//!     clock,
+//!     entrypoint::ProgramResult,
+//!     msg,
+//!     pubkey::Pubkey,
+//!     sysvar::Sysvar,
+//! };
+//!
+//! fn process_instruction(
+//!     program_id: &Pubkey,
+//!     accounts: &[AccountInfo],
+//!     instruction_data: &[u8],
+//! ) -> ProgramResult {
+//!     let clock = clock::Clock::get()?;
+//!     msg!("clock: {:#?}", clock);
+//!     Ok(())
+//! }
+//! ```
+//!
+//! Some sysvars are too large to deserialize within a program, and
+//! `Sysvar::from_account_info` returns an error. Some sysvars are too large
+//! to deserialize within a program, and attempting to will exhaust the
+//! program's compute budget. Some sysvars do not implement `Sysvar::get` and
+//! return an error. Some sysvars have custom deserializers that do not
+//! implement the `Sysvar` trait. These cases are documented in the modules for
+//! individual sysvars.
+//!
+//! For more details see the Solana [documentation on sysvars][sysvardoc].
+//!
+//! [sysvardoc]: https://docs.solana.com/developing/runtime-facilities/sysvars
+//!
+//! # Limitations to the Solana version of the Rust standard library
```

Comment:
Would be best just to refer to docs.solana.com.  Any suggestions you have to improve the docs would be appreciated, the doc src files are here: https://github.com/solana-labs/solana/tree/master/docs/src